api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/prev/CompletedIsoniazidProphylaticTreatmentCalculation.java

```diff
@@ -0,0 +1,183 @@
+            hivMetadata.getDataFinalizacaoProfilaxiaIsoniazidaConcept(),
+            null,
+            location,
+            false,
+            completionPeriodStartDate,
+            completionPeriodEndDate,
+            cohort,
+            context);
+    CalculationResultMap isoniazidUsageObservationsList =
+        ePTSCalculationService.allObservations(
+            hivMetadata.getIsoniazidUsageConcept(),
+            hivMetadata.getYesConcept(),
+            consultationEncounterTypes,
+            location,
+            cohort,
+            context);
+
+    for (Integer patientId : cohort) {
+      Obs startProfilaxiaObs =
+          EptsCalculationUtils.resultForPatient(startProfilaxiaObservations, patientId);
+      Obs endProfilaxiaObs =
+          EptsCalculationUtils.resultForPatient(endProfilaxiaObservations, patientId);
+      Date startDate = getDateFromObs(startProfilaxiaObs);
+      Date endDate = getDateFromObs(endProfilaxiaObs);
+      boolean inconsistent =
+          (startDate != null && endDate != null && startDate.compareTo(endDate) > 0)
+              || (startDate == null && endDate != null);
+      if (!inconsistent && startDate != null) {
+        boolean completed = startDate != null && endDate != null;
+        if (completed) {
+          int profilaxiaDuration =
+              Days.daysIn(new Interval(startDate.getTime(), endDate.getTime())).getDays();
+          if (profilaxiaDuration >= MINIMUM_DURATION_IN_DAYS) {
+            map.put(patientId, new BooleanResult(true, this));
+          }
+        } else {
+          int yesAnswers =
+              calculateNumberOfYesAnswers(isoniazidUsageObservationsList, patientId, startDate);
+          if (yesAnswers >= NUMBER_ISONIAZID_USAGE_TO_CONSIDER_COMPLETED) {
+            map.put(patientId, new BooleanResult(true, this));
+          }
+        }
+      }
+    }
+    return map;
+  }
+
+  private int calculateNumberOfYesAnswers(
+      CalculationResultMap isoniazidUsageObservationsList, Integer patientId, Date startDate) {
+    List<Obs> isoniazidUsageObservations =
+        EptsCalculationUtils.extractResultValues(
+            (ListResult) isoniazidUsageObservationsList.get(patientId));
+    int count = 0;
+    Date isoniazidUsageEndDate = getIsoniazidUsageEndDate(startDate);
+    for (Obs obs : isoniazidUsageObservations) {
+      Date date = obs.getObsDatetime();
+      if (date.compareTo(startDate) >= 0 && date.compareTo(isoniazidUsageEndDate) <= 0) {
+        count++;
+      }
+    }
+    return count;
+  }
+
+  private Date getIsoniazidUsageEndDate(Date startDate) {
+    Calendar calendar = Calendar.getInstance();
+    calendar.setTime(startDate);
+    calendar.add(Calendar.MONTH, MONTHS_TO_CHECK_FOR_ISONIAZID_USAGE);
+    return calendar.getTime();
+  }
+
+  private Date getDateFromObs(Obs obs) {
+    if (obs != null) {
+      return obs.getValueDatetime();
+    }
+    return null;
+  }
+
+  private Date getDateParameter(
```

Comment:
Which date parameter are you trying to get here? endDate or startDate or this something specific for this calculation? we might have those be taken care of by the calculation cohort definition evaluator such that those can be used by other calculations as well. Please check if that already exisits in the evaluator

Reply:
The for dates from reportingPeriod and previousPeriod